packages/store/src/tightcoder/DecodeSlice.ejs

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { TightCoder } from "./TightCoder.sol";
+import { Slice } from "../Slice.sol";
+import { SchemaType } from "../Types.sol";
+
+library DecodeSlice {
+  /************************************************************************
+   *
+   *    uint8 - uint256
+   *
+   ************************************************************************/
+<% for (let i = 8; i <= 256; i += 8) { -%>
+
+  function decodeArray_uint<%= i %>(Slice self) internal pure returns (uint<%= i %>[] memory output) {
```

Comment:
ooc why did you decide for this naming format (`decodeArray_uint256`) instead of something like `decodeUint256Array`? Also, since the library is called `DecodeSlice`, I would have expected the function to be called something like `toUint256Array`, but no strong opinion.

Reply:
In short I'm going for `<method>_<T>` as way to resemble generics

> why did you decide for this naming format (decodeArray_uint256) instead of something like decodeUint256Array?

The main reason is autogen. After _ comes the exact type name. I'd rather not do more manipulations than necessary since this is already quite complicated.

> Also, since the library is called DecodeSlice, I would have expected the function to be called something like toUint256Array

Similar reason - I want the "method name" to be purely as a prefix, and "type name" purely as a postfix, both separated by `_`.
For `encode`/`decode` - I feel like while writing it I had a harder time remembering which methods did what with `to`/`from`. With decode I had a better analogy to abi.decode etc.